### PR TITLE
Creating functions to rename application pools and websites.

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Carbon.IIS/Functions/Rename-CIisAppPool.ps1 [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    $appPool = Get-CIisAppPool -Name $Name
+
+    if (-not $appPool)
+    {
+        return
+    }
+
+    $appPoolCount = ($appPool | Measure-Object).Count
+    if ($appPoolCount -gt 1)
+    {
```
Comment (KhoiKy) [inline Carbon.IIS/Functions/Rename-CIisAppPool.ps1:46]: I think this outer if statement isn't needed? There's also an if statement identical to this one nested.

File: Carbon.IIS/Functions/Rename-CIisAppPool.ps1 [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    $appPool = Get-CIisAppPool -Name $Name
+
+    if (-not $appPool)
+    {
+        return
+    }
+
+    $appPoolCount = ($appPool | Measure-Object).Count
+    if ($appPoolCount -gt 1)
```
Comment (KhoiKy) [inline Carbon.IIS/Functions/Rename-CIisAppPool.ps1:45]: Should this be -ge 1?

File: Carbon.IIS/Functions/Rename-CIisAppPool.ps1 [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        $msg = 'is 1 application pool that matches'
+        if ($appPoolCount -gt 1)
```
Comment (KhoiKy) [inline Carbon.IIS/Functions/Rename-CIisAppPool.ps1]: This inner if statement isn't needed? Only an issue when there are more than one app pool matching the Name specified. So error message should always be "are...$appPoolCount"

File: Carbon.IIS/Functions/Rename-CIisWebsite.ps1 [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    if (-not $site)
+    {
+        return
+    }
+
+    $siteCount = ($site | Measure-Object).Count
+    if ($siteCount -gt 1)
+    {
+        $msg = 'is 1 website that matches'
+        if ($siteCount -gt 1)
```
Comment (KhoiKy) [inline Carbon.IIS/Functions/Rename-CIisWebsite.ps1]: Same here, I think this inner if statement isn't needed.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,8 +3,18 @@
 
 # Carbon.IIS Changelog
 
+## 1.5.0
+
+### Added
+
+* `Rename-CIisAppPool` to rename IIS application pools.
+* `Renaem-CIisWebsite` to rename IIS websites.
```
Comment (KhoiKy) [inline CHANGELOG.md]: Typo

**Review by KhoiKy — CHANGES_REQUESTED** [review]
Comments in files.

File: Tests/Rename-CIisAppPool.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+        $mgr = [Microsoft.Web.Administration.ServerManager]::New()
+        $appPool = $mgr.ApplicationPools | Where-Object 'name' -EQ $Named
+        if ($Not -and $Exists)
+        {
+            $appPool | Should -BeNullOrEmpty
+        }
+        elseif ($Exists)
+        {
+            $appPool | Should -Not -BeNullOrEmpty
```
Comment (decoyjoe) [inline Tests/Rename-CIisAppPool.Tests.ps1:40]: I think you can eliminate the conditional by just doing:

```powershell
$appPool | Should -Not:$Not -BeNullOrEmpty
```

**Review by decoyjoe — APPROVED** [review]

File: Carbon.IIS/Functions/Rename-CIisAppPool.ps1 (hunk 3 above)
Comment (splatteredbits) [reply Carbon.IIS/Functions/Rename-CIisAppPool.ps1]: ZOMFSM. 🙃

File: Tests/Rename-CIisAppPool.Tests.ps1 (hunk 6 above)
Comment (splatteredbits) [reply Tests/Rename-CIisAppPool.Tests.ps1:40]: Agreed, but this is a conscience style choice for me: I like my tests and assertions to read like a sentence and having optional/conditional parameters breaks that flow.

**Review by KhoiKy — APPROVED** [review]

**Pull request merged**